### ISSUE TITLE
Can't translate the calendar language values by overrides

### ISSUE THREAD

**Issue by Mausino** [joomla/joomla-cms#20296]
### Steps to reproduce the issue

I want custom values for calendar pop up system window in own region language

### Expected result

Can add it via overrides, because i must create js file in system

media/system/js/fields/calendar-locales/en.js 
or
media/system/js/fields/calendar-locales/de.js or

### Actual result
My employees can't control language on all site.. need IT to solve some problems with tranlations

Joomla 3.8.7

**Comment by zero-24** [comment]
What kind of problems do you have with the german or original translations?

Maybe we can fix them upstream so no need to override that yourself.

Iirc there is also a way to override the files in media but I don't know it by heart need to check that later if not somone else beat me with it ;-)

**Comment by brianteeman** [comment]
I think this is the post you are referring to about overriding /media

https://www.babdev.com/blog/139-use-the-media-folder-allow-overridable-media

**Comment by Mausino** [comment]
@zero-24 i don't have problem with original translations... but we creating the region languages which are many differencies between original language and region/folk language

My question is: why can't be by overrides tranlate all words (text) in whole joomla ? 

@brianteeman i didn't know before about this article.. but it helps a lot :) thank you

**Comment by infograf768** [comment]
> My question is: why can't be by overrides tranlate all words (text) in whole joomla ?

question to @dgrammatiko 
The simple reply is that these lang values are javascript, (would be a similar reply for TinyMce lang values).

**Comment by dgrammatiko** [comment]
It **should** be overridable through the template, if it’s not then I wrongly hardcoded it. But using php (cpu cycles) to generate json files is pretty stupid and wrong pattern imho

**Comment by brianteeman** [comment]
> @zero-24 i don't have problem with original translations... but we creating the region languages which are many differencies between original language and region/folk language

You should be doing this as complete language files then and not overrides - and it will make them reusable across more than one site

**Comment by dgrammatiko** [comment]
@infograf768 @Mausino I've just checked the code and **all** the assets are overridable : https://github.com/joomla/joomla-cms/blob/e99b38dc22fc61ebaa1384cda2c3a60379a1383b/layouts/joomla/form/field/calendar.php#L101-L104

Just a bit of the history why we did it with a static file instead of PHP:
- we wanted to eliminate the risk of unusable calendar due to missing strings
- performance, wrote about it in the previous comment as well

So although this doesn't follow the usual workflow for overriding lang strings, it is still possible, all you need to do is copy the files you need to override from `media\system\fields\calendar\calendar-locales\[language code].js` to your template's js folder and then edit the strings to fit your need

**Comment by mbabker** [comment]
It looks like it's still somewhat dependent on the thing you want to override existing in the `media` directory first:

https://github.com/joomla/joomla-cms/blob/e99b38dc22fc61ebaa1384cda2c3a60379a1383b/libraries/joomla/form/fields/calendar.php#L280-L294

If you for whatever reason wanted to have separate configs for en-GB and en-US you'd still have to create `en-gb.js` and `en-us.js` in `/media/system/js/fields/calendar-locales` to let a template override take effect.

So you CAN override an existing file, but if you needed to add a new locale for whatever reason, you'd have to get it into the core directory first.

**Comment by infograf768** [comment]
> So you CAN override an existing file, but if you needed to add a new locale for whatever reason, you'd have to get it into the core directory first.

I propose this:
It is possible to implement for a new language pack (that would not be available through J or which js files are not yet in core) the possibility to upload/get the specific files from `media/xx-XX/js/fields/calendar-locales/` . These could also be used as overrides when they exist in core.

I have tested this for en-US and it works great:
```
		if (is_file(JPATH_ROOT . '/media/' . $tag . '/js/fields/calendar-locales/' . strtolower($tag) . '.js'))
		{
			$localesPath = $tag . '/fields/calendar-locales/' . strtolower($tag) . '.js';
		}
```
The advantage of such a solution is that it would not be deleted by a J update. It can be created manually or installed by a lang pack.

@mbabker @dgrammatiko 
PR proposed here: https://github.com/joomla/joomla-cms/pull/20303

**Comment by ghost** [comment]
closed as having Pull Request #20303.
